virtcontainers/sandbox.go

```diff
@@ -1073,13 +1079,59 @@ func (s *Sandbox) addContainer(c *Container) error {
 	}
 	s.containers[c.id] = c
 
-	ann := c.GetAnnotations()
-	if ann[annotations.ContainerTypeKey] == string(PodSandbox) {
-		s.state.CgroupPath = c.state.CgroupPath
-		if !s.supportNewStore() {
-			return s.store.Store(store.State, s.state)
+	if !s.supportNewStore() {
+		return s.store.Store(store.State, s.state)
+	}
+
+	return nil
+}
+
+// Look in all container configurations a container of type PodSandbox
+// use the cgroup of the PodSandbox as Sandbox Cgroup.
+func (s *Sandbox) retrieveCgroupPath() error {
+	for _, cConfig := range s.config.Containers {
+
+		ann := cConfig.Annotations
+		if ann[annotations.ContainerTypeKey] != string(PodSandbox) {
+			continue
+		}
+
+		config, ok := ann[annotations.ConfigJSONKey]
+		if !ok {
+			return fmt.Errorf("Could not find json config in annotations")
+		}
+
+		var spec specs.Spec
+		if err := json.Unmarshal([]byte(config), &spec); err != nil {
+			return err
+		}
+
+		validContainerCgroup := utils.ValidCgroupPath(spec.Linux.CgroupsPath)
+		if !s.config.SandboxCgroupOnly {
+			// Use the same cgroup of the container sandbox as the sandbox cgroup
+			s.state.CgroupPath = validContainerCgroup
+			return nil
 		}
+
+		// Use the parent cgroup of the container sandbox as the sandbox cgroup
+		s.state.CgroupPath = filepath.Dir(validContainerCgroup) + "/kata-sandbox-" + cConfig.ID
```

Comment:
Rather than using `+`, it would be better to use `filepath.Join()`. Also, that fixed string could be a top-level `const` maybe.